frontend/ui/widget/inputdialog.lua

```diff
@@ -464,9 +464,14 @@ function InputDialog:init()
 
     -- insert check button before the regular buttons
     if self.check_button then
-        local nr_elements = #self.dialog_frame[1]
-        table.insert(self.dialog_frame[1], nr_elements-1, vspan_before_input_text)
-        table.insert(self.dialog_frame[1], nr_elements-1, self.check_button)
+        local nb_elements = #self.dialog_frame[1]
+        table.insert(self.dialog_frame[1], nb_elements-1, CenterContainer:new{
+                dimen = Geom:new{
+                    w = self.width,
+                    h = self._input_widget:getSize().h,
```

Comment:
Is this the right way to compute its height ? Or you were just lucky that it gives you a height that looked just fine ? :)

Reply:
I will pull out the checkbuttons from inputdialog and add them manualy in readersearch. I am almost ready. But it would be great if you could look at that solution (I am not quite firm with the UI dimension calcs.)

Reply:
In general (there are quirky exceptions) the result of `getSize()` is what ends up in the `dimen` field after `paintTo` (which is where this sort of stuff happens).

There are a few precedents in the code for doing that by hand before `paintTo`, IIRC.

Whether that's actually the best solution here is another thing entirely ;).